src/components/sketches/Scene5_happy.ts

```diff
@@ -0,0 +1,266 @@
+    private _firingParticle: Particle;
+    private _particles: Array<Particle>;
+
+    constructor() {
+      this.explodeSize = p.random(0, EXPLODE_RANGE);
+      this._firingParticle = new Particle(p.random(p.width), p.height, true);
+      this.exploded = false;
+      this._particles = [];
+    }
+
+    public update() {
+      if (!this.exploded) {
+        this._firingParticle.update(gravity);
+
+        if (this._firingParticle.vel.y >= 0) {
+          this.explode();
+        }
+      }
+
+      this._particles = this._particles.filter((particle) => {
+        particle.update(gravity);
+        return !particle.done();
+      });
+    }
+
+    public done() {
+      return this.exploded && this._particles.length === 0;
+    }
+
+    public show() {
+      if (!this.exploded) {
+        this._firingParticle.show();
+      }
+
+      for (let i = 0; i < this._particles.length; i++) {
+        this._particles[i].show();
+      }
+    }
+
+    private explode() {
+      this.exploded = true;
+      for (let i = 0; i < this.explodeSize * EXPLODE_OFFSET; i++) {
+        const particle = new Particle(
+          this._firingParticle.pos.x,
+          this._firingParticle.pos.y,
+          false,
+        );
+        this._particles.push(particle);
+      }
+    }
+  }
+
+  class Particle {
+    pos: p5.Vector;
+    public isFirework: boolean;
+    lifespan: number;
+    private _acc: p5.Vector;
+    rand: number;
+    vel: p5.Vector;
+
+    constructor(x: number, y: number, isFirework: boolean) {
+      this.pos = p.createVector(x, y);
+      this.isFirework = isFirework;
+      this.lifespan = FIREWORK_LIFESPAN;
+      this._acc = p.createVector(0, 0);
+      this.rand = p.random(0, 3);
+      this.vel = this.isFirework
+        ? p.createVector(0, p.random(-12, -8))
+        : p5.Vector.random2D().mult(p.random(2, 10));
+    }
+
+    public update(force: p5.Vector) {
+      if (!this.isFirework) {
+        this.vel.mult(0.9);
+        this.lifespan -= 4;
+      }
+      this._acc.add(force);
+      this.vel.add(this._acc);
+      this.pos.add(this.vel);
+      this._acc.mult(0);
+    }
+
+    public done() {
+      return this.lifespan < 0;
+    }
+
+    public show() {
+      if (!this.isFirework) {
+        /** after explosion */
+        p.strokeWeight(6);
+        p.stroke(255, 211, 97, this.lifespan);
+      } else {
+        p.strokeWeight(8);
+      }
+
+      const opacity = this.rand < 1 ? 0.2 : this.rand < 2 ? 0.4 : 0.6;
+      p.stroke(`rgba(255,211,97,${opacity})`);
+      p.point(this.pos.x, this.pos.y);
+    }
+  }
+
+  class RingParticle {
+    private _pos: p5.Vector;
+    private _vel: p5.Vector;
+    private _acc: p5.Vector;
+    private _radius: number;
+    private _time: number;
+    private _explode: boolean;
+    private _lifespan: number;
+    private _history: p5.Vector[];
+
+    constructor(x: number, y: number, theta: number) {
+      /** Position */
+      this._pos = p.createVector(x, y);
```

Comment:
So you would actually put the comment at the declaration!

```suggestion
    /** The position of the Ring Particle */
    private _pos: p5.Vector;
    private _vel: p5.Vector;
    private _acc: p5.Vector;
    private _radius: number;
    private _time: number;
    private _explode: boolean;
    private _lifespan: number;
    private _history: p5.Vector[];

    constructor(x: number, y: number, theta: number) {
      this._pos = p.createVector(x, y);
```